hack/create-identity-secret.sh

```diff
@@ -28,8 +28,17 @@ make --directory="${REPO_ROOT}" "${KUBECTL##*/}"
 # shellcheck source=hack/parse-prow-creds.sh
 source "${REPO_ROOT}/hack/parse-prow-creds.sh"
 
-if ! "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null; then
+if ! "${KUBECTL}" get secret "${AZURE_CLUSTER_IDENTITY_SECRET_NAME}" > /dev/null && [[ -v "AZURE_CLIENT_SECRET" ]]; then
```

Comment:
why "-v" and not "-n" ?

Reply:
I think both would be practically equivalent here. From bash(1):
```
-v varname
              True if the shell variable varname is set (has been assigned a value).
-n string
              True if the length of string is non-zero.
```

Reply:
right, the edge case is if `$AZURE_CLIENT_SECRET` is set to an empty string then we are going to make a non-functional secret

this is probably academic, but that's why `-v` is probably not sufficient for what we're actually doing w/ the value of `$AZURE_CLIENT_SECRET`